app/models/assignee.rb

```diff
@@ -0,0 +1,109 @@
+# encoding: UTF-8
+
+# An assignee describes a user and have some common attributes.
+# The assignee is used as a Value Object, and is not saved with ActiveRecord.
+# For example - see implementation in CafeWork model.
+# An assignee can only be created with Assignee.new(param_hash), not create.
+# You cannot save values here, that belongs in the ActiveRecord model.
+# Ref: http://blog.codeclimate.com/blog/2012/10/17/7-ways-to-decompose-fat-activerecord-models/
+# /d.wessman
+class Assignee
+  include ActiveModel::Model
+
+  # Attributes that can be set and be read.
+  attr_accessor :name, :lastname, :phone, :email, :profile_id, :access_code, :profile
+
+  # Method used to setup new Assignee with a user > profile/access_code
+  # User present > Profile is set to User.profile
+  # User absent  > Access Code is generated
+  def self.setup(assignee_params, user)
+    new(assignee_params).prepare(user)
+  end
+
+  # To define if the Assignee is set and present.
+  def is_present?
+    has_profile? || has_access_code?
+  end
+
+  # Printing methods
+  def p_name
+    %(#{name} #{lastname})
+  end
+
+  # Can be used in mailer, will generate proper tag.
+  def p_email
+    %("#{name} #{lastname}" <#{email}>)
+  end
+
+  # A way to clear all the attributes of the Assignee
+  # Returns array of attributes, this way it can be used like:
+  # Model.attributes = assignee.clear_attributes
+  # Which saves a lot of rows and declarations.
+  def clear_attributes
+    self.name, self.lastname, self.phone = nil, nil, nil
+    self.email, self.profile_id, self.access_code = nil, nil, nil
+    attributes
+  end
+
+
+  def load_profile(user)
+    if (user.present?) && !has_worker?
+      self.name = user.profile.name
+      self.lastname = user.profile.lastname
+      self.email = user.profile.email
+      self.phone = user.profile.phone
+    end
+    attributes
+  end
+
+  # Returns a hash of the current attributes
+  def attributes
+    {
+        name: name, lastname: lastname, email: email,
```

Comment:
Use 2 spaces for indentation in a hash, relative to the start of the line where the left curly brace is.